.gitignore

```diff
@@ -199,3 +200,6 @@ data/**
 
 # PyCharm
 .idea/
+
+requirements-dev.lock
```

Comment:
Module lock files should generally not be ignored, as e.g. discussed [here](https://stackoverflow.com/questions/48524417/should-the-package-lock-json-file-be-added-to-gitignore).
I think these are `rye`-specific, so I can't find a lot of info on them, but we don't want our dependencies to ever be different, therefore the lock files should be version-controlled.